Puzzle/src/org/fundacionjala/devops/bootcamp/Move/Movement.java

```diff
@@ -0,0 +1,182 @@
+    public String posibleMovements(int[][] board) {
+        String movements = "You can go to: ";
+        Position newPosition= new Position();
+        String zeroPosition= String.valueOf(newPosition.getHolePosition());
+        int boardSize = board.length;
+        for (int i = 0; i < board.length; i++) {
+            for (int j = 0; j < board[i].length; j++) {
+                if (board[i][j] == 0) {
+                    System.out.println("The position of the zero is: " + (i+1) + "," + (j+1));
+                    if (i == 0) {
+                        movements += "S ";
+                    }else if (i == boardSize - 1) {
+                        movements += "N ";
+                    }else{
+                        movements += "N, S ";
+                    }
+                    if (j == 0) {
+                        movements += ", E ";
+                    }else if (j == boardSize - 1) {
+                        movements += ", O ";
+                    }else{
+                        movements += ", O, E ";
+                    }
+                }
+            }
+        }
+        System.out.println(movements);
+        return movements;
+    }
+    //Method to move the zero, that receive the board and the movement("N", "S", "E" or "O"):
+    public int moveZero(int[][] board, String movement) {
+        String holePosition = findZero(board);
+        size= board.length;
+        if(isValidMovement(holePosition, movement,size)) {
+            if(movement.equals("N")){
+                NorthMove N = new NorthMove();
+                N.execute(board);
+                return 1;
+            }else if(movement.equals("S")){
+                SouthMove S = new SouthMove();
+                S.execute(board);
+                return 1;
+            }else if(movement.equals("E")){
+                EastMove E = new EastMove();
+                E.execute(board);
+                return 1;
+            }else if(movement.equals("O")){
+                WestMove O = new WestMove();
+                O.execute(board);
+                return 1;
+            }
+        }
+        return 0;
+    }
+
+    //Check which is the position of the 0 in the board:
+    public String findZero(int[][] board) {
+        String position = "";
+        for (int i = 0; i < board.length; i++) {
+            for (int j = 0; j < board[i].length; j++) {
+                if (board[i][j] == 0) {
+                    position+=i+","+j;
+                }
+            }
+        }
+        return position;
+    }
+
+    public String randomMovement() {
+        String[] movements = {"N", "S", "E", "O"};
+        int random = (int) (Math.random() * movements.length);
+        return movements[random];
+    }
+
+    public void rollBackBoard(int[][] board,String commands) {
+        String[] commandsArray = commands.split(" ");
+        for (int i = 0; i < commandsArray.length; i++) {
+            if(commandsArray[i].equals("N")){
```

Comment:
I noticed that you are using Command pattern, is it something that we can use here to avoid the if-else block?